File: L2/S4/I41-Algo/TP12-lexicographique/ordre.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

int ordre(char *u, char *v) {
    int i, len_u = strlen(u), len_v = strlen(v);
    for(i = 0; i <= len_u && i <= len_v && u[i] == v[i]; i++);
    if(i > len_u && i > len_v)
        return 0;
    else if(i > len_u || u[i] < v[i]) {
        printf("%d %d\n", u[i], v[i]);
        return -1;
    }
    else if(i > len_v || v[i] < u[i]){
        printf("%d %d\n", u[i], v[i]);
        return 1;
    }
}

void main(int argc, char **argv) {
    int d = ordre(argv[1], argv[2]);
    if (d == -1)
        printf("%s < %s\n",argv[1],argv[2]);
    if (d == 0)
        printf("%s == %s\n",argv[1],argv[2]);
    if (d == 1)
        printf("%s > %s\n",argv[1],argv[2]);
}

File: L2/S4/I41-Algo/TP12-lexicographique/ordre.py
#!/usr/bin/python3
from sys import argv
def ordre(u, v):
    i = 0
    while (i < len(u) and i < len(v) and u[i] == v[i]):
        i += 1

    if(i == len(u) and i == len(v)):
        return 0
    elif(i == len(u) or (i<len(v) and v[i] > u[i])):
        return -1
    else:
        return 1

print(argv[0])

d = ordre(argv[1], argv[2])
if (d == -1):
    print(argv[1],"<",argv[2]);
if (d == 0):
    print(argv[1],"==",argv[2]);
if (d == 1):
    print(argv[1],">",argv[2]);
